### DISCUSSION TITLE
Does htmlNode only extend certain document object functions?

### DISCUSSION THREAD

**Discussion by psheehymn** [gapitio/gapit-htmlgraphics-panel#160] [Q&A] [answered]
I am attempting to update the font size of all the table cells using a particular class, so I tried to use the **document** object method `.getElementsByClassName()` so I could loop over all the cells. Obviously substituting **htmlNode** for **document**, I got the following error from the panel: 

> Error executing onInit
> htmlNode.getElementsByClassName is not a function

Does **htmlNode** only support a subset of the **document** object methods?

**Comment by ZuperZee** [comment, marked as answer]
htmlNode is a ShadowRoot <https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot>
It inherets DocumentFragment <https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment>
![image](https://github.com/gapitio/gapit-htmlgraphics-panel/assets/39485579/97f1b545-24b1-4752-b4ec-50886ae521c0)

To get classNames from the htmlNode you can use `querySelectorAll` from the DocumentFragment <https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment/querySelectorAll>
```js
htmlNode.querySelectorAll(".class-name")
```

Or use a child element that has the `getElementsByClassName` function <https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByClassName>.
```js
htmlNode.lastChild.getElementsByClassName("class-name")
```

**Reply by psheehymn** [reply]
`htmlNode.querySelectorAll(".class-name")` worked perfectly. I also now have a better understanding of where to look for supporting information. Thanks for both!
